src/url-params.ts

```diff
@@ -163,10 +163,20 @@ export class AnalyticsUrlParams {
 	/**
 	 * @return full query parameter string that can be appended to URLs
 	 */
-	getQueryString(): string {
-		return [this.getDeviceIdsQueryString(), this.getSessionIdQueryString()]
-			.filter(x => x)
-			.join('&');
+	getQueryString(destinationHostname?: string): string {
+		const regex = /([a-zA-Z0-9-]+)(\.[a-zA-Z]{2,5})?(\.[a-zA-Z]+$)/g;
```

Comment:
Good stuff with this regex here. It works with all cases I could imagine on balena websites. Then it returns `test.test.io` as is.

```js
(new URL('https://test.test.io')).hostname.match(regex) -> test.test.io
```

Can you please expand the reasoning behind the numbers in `{2,5}`? 

Setting it to `{2,3}` makes my test case return `test.io` and doesn't break other tests. That's why I'd like to understand this 5 number.

Reply:
I was using 5 because it was to handle those unusual TDL like ".forum", ".audio" etc. But I didnt noticed that it could missinterpret short domain names like "test". I switched to 3 now assuming we wont be using weird long TDLs for the moment.  Because otherwise, things like "hub.balena.edge.io" where hub is a subdomain, balena a domain, and .edge.io TLD , it matches only "edge.io"

Reply:
Got it. 

> assuming we wont be using weird long TDLs

That's fair. Can you please comment this assumption in code Ezequiel?

Additionally, it'd be great to have unit tests here that shows the expected values for such transformation. Perhaps one of the tests could be `hub.balena.edge.io -> edge.io` to illustrate your assumption and the expected behavior.

Reply:
I added the assumption on both the code and on the test to explicitly state what should happen and why